src/cplscheme/BaseCouplingScheme.hpp

```diff
@@ -262,9 +258,16 @@ class BaseCouplingScheme : public CouplingScheme {
    *
    * @param m2n M2N used for communication
    * @param receiveData DataMap associated with received data
-   * @param initialCommunication if true and exchange of substeps is deactivated, will store received data for WINDOW_START and WINDOW_END, else store received data only for WINDOW_END
    */
-  void receiveData(const m2n::PtrM2N &m2n, const DataMap &receiveData, bool initialCommunication = false);
+  void receiveData(const m2n::PtrM2N &m2n, const DataMap &receiveData);
+
+  /**
+   * @brief Like receiveData, but sets window time to end of window. Needed by SerialCouplingScheme.
+   *
+   * @param m2n M2N used for communication
+   * @param receiveData DataMap associated with received data
+   */
+  void receiveDataForWindowEnd(const m2n::PtrM2N &m2n, const DataMap &receiveData);
```

Comment:
Couldn't we move this function to `SerialCouplingScheme`?

Could you please write more in the comment on why this function is needed? This one is a bit tricky, we will wonder in the future.

Reply:
Sure. But then `BaseCouplingScheme::_computedTimeWindowPart` needs a protected setter to allow `SerialCouplingScheme` to access it. Not sure that this really simplifies things. I would rather keep it as it is, because it's less risky to expose `receiveDataForWindowEnd` to `ParallelCouplingScheme` than exposing `_computedTimeWindowPart` to all derived coupling schemes.

Reply:
I added some more information on the function.